### DISCUSSION TITLE
What is the end of life for CF V2 APIs?

### DISCUSSION THREAD

**Discussion by karthik-sub** [cloudfoundry/cloud_controller_ng#2725] [General]
Question: We are using CF V2 APIs in our project. Given that V2 is deprecated we would like to know the end of life for V2 APIs? This info will help us to plan the upgrade accordingly. Thank you.

**Comment by Gerg** [comment]
I converted this to a discussion instead of an issue.

**Comment by Gerg** [comment]
In short, there are currently no plans to disable or delete the v2 API. 

Based on our analysis, the effort to migrate all clients off of the v2 API is too large for the engineering resources that are currently available. We have successfully migrated a number of major open-source and proprietary clients to v3, but there are still numerous v2 clients that are in common use but are not actively maintained. 

Since the v2 API is deprecated, it will not receive new features or bug fixes. As such, we recommend that client authors migrate to the v3 API as soon as possible.

**Comment by philippthun** [comment]
There is a [configuration option](https://github.com/cloudfoundry/capi-release/blob/e119b9360870171aa1b52c750763b065b11d629b/jobs/cloud_controller_ng/spec#L387-L389) to disable v2 endpoints. So even though they might still be supported by code, platform operators can decide to stop support in a specific foundation.

**Comment by Gerg** [comment]
Further discussion in the RFC: https://github.com/cloudfoundry/community/pull/941

**Comment by stephanme** [comment]
[RFC-0032 CF API v2 End of Life](https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0032-cfapiv2-eol.md) was accepted by the TOC.

tl;dr;
Predicted end-of-life for CF API v2 is mid/end 2026. Refer to the RFC for the complete roadmap which contains TOC checkpoints before CF API v2 finally gets removed from CAPI and cf-deployment.
